### PR TITLE
fix IPv6 syslog messages handled

### PR DESCRIPTION
**Pull request by cborla**
|Related issue|Manual Testing|
|---|---|
|https://github.com/wazuh/wazuh/issues/13708|https://github.com/wazuh/wazuh-qa/issues/3094|

## Description

Syslog messages received by the Wazuh Syslog server (remoted) via IPv6 are correctly handled, the `location` and `full_log` fields are correctly parsed (splitted).

## Use case 

Remote Syslog messages are received on the Wazuh manager, from a device which address is of the IPv6 format.

### Configuracion

```xml
  <global>
       ...
       <logall_json>yes</logall_json>
       ...
  </global>
  
  <remote>
    <connection>syslog</connection>
    <port>5050</port>
    <protocol>tcp</protocol>
    <ipv6>yes</ipv6>
    <allowed-ips>any</allowed-ips>
  </remote>
```
## Tests

Send a Syslog message from an IPv6 address (in this case, through the loopback interface).

```bash
echo "IPv6 test msg" | nc -v -w 0 -6 "::1"  5050
```
or running wazuh-logtest
```bash
/var/ossec/bin/wazuh-logtest -l "0000:0000:0000:0000:0000:0000:0000:0001" -d
```

Result

```json
{
    "timestamp": "2022-06-03T16:50:16.660+0000",
    "agent": {
        "id": "000",
        "name": "200-u20-dev-manager"
    },
    "manager": {
        "name": "200-u20-dev-manager"
    },
    "id": "1654275016.1163645",
    "full_log": "IPv6 test msg",
    "decoder": {},
    "location": "0000:0000:0000:0000:0000:0000:0000:0001"
}
```
Testing ipv4

```bash
echo "IPv4 test msg" | nc -v -w 0 localhost 505
```
```json
{
    "timestamp": "2022-06-03T16:50:16.660+0000",
    "agent": {
        "id": "000",
        "name": "200-u20-dev-manager"
    },
    "manager": {
        "name": "200-u20-dev-manager"
    },
    "id": "1654275016.1163645",
    "full_log": "IPv4 test msg",
    "decoder": {},
    "location": "127.0.0.1"
}
```

- Compilation without warnings in every supported platform
  - [x] Linux
- [x] Source installation
- [x] Package installation
- [x] Source upgrade
- [x] Package upgrade
- [x] Review logs syntax and correct language

- Memory tests for Linux
  - [ ] Coverity
  - [x] Valgrind (memcheck and descriptor leaks check)

- [x] Retrocompatibility with older Wazuh versions
- [x] The data flow works as expected (agent-manager-api-app)
- [x] Added unit tests (for new features)

### CODE REVIEW

**Review by TomasTurina — APPROVED** [review]

File: src/analysisd/cleanevent.c [hunk 1]
```diff
@@ -26,27 +28,43 @@ static const char *(month[]) = {"Jan", "Feb", "Mar", "Apr", "May", "Jun",
...
+    } else if (SYSLOG_MQ == id && msg[4] == ':' && strlen(msg) > IPV6_LAST_DIGIT
+              && msg[9] == ':' && msg[14] == ':' && msg[19] == ':' && msg[24] == ':'
+              && msg[29] == ':' && msg[34] == ':') {
+
+        src_msg = msg + IPV6_LAST_DIGIT;
     }
 
-    pieces = strchr(arrow ? arrow : msg, ':');
+    pieces = strchr(src_msg ? src_msg : msg, ':');
     if (!pieces) {
```
[the remark this answers is not in the document]
Comment (cborla) [reply src/analysisd/cleanevent.c]: Last check should be 
`if (pieces == NULL) {`

**Review by TomasTurina — APPROVED** [review]

**Review by juliancnn — APPROVED** [review]

**Comment by jmv74211** [comment]
## QA review

- **Type**: Manual testing.
- **Status**: Blocked  🟡
- **Testing issue**: https://github.com/wazuh/wazuh-qa/issues/3094
- **Comments**: The use cases related to the changes in this PR have been tested and verified that everything works correctly.
  
  There is a check related to `Test: AR IT - Tier 0 - windows_agent` which is red and should be fixed. The problem related to that check is reported in this issue https://github.com/wazuh/wazuh/issues/13077.
  
  For now the PR is blocked until the necessary fixes are made, unless otherwise indicated by priorities.

File: src/analysisd/cleanevent.c [hunk 2]
```diff
@@ -26,28 +28,43 @@ static const char *(month[]) = {"Jan", "Feb", "Mar", "Apr", "May", "Jun",
...
     msg += 2;
 
     /* Avoid ipv6 ':', msg that include "[" have an "->" after the ip */
     if (*msg == '[') {
-        if (!(arrow = strstr(msg, "->"))) {
+        if (src_msg = strstr(msg, "->"), src_msg == NULL) {
             merror(FORMAT_ERROR);
             return (-1);
         }
+    } else if (SYSLOG_MQ == id && msg[4] == ':' && strlen(msg) > IPV6_LAST_DIGIT
```
Comment (TomasTurina) [inline src/analysisd/cleanevent.c:60]: I think we should check the size before looking at the first `:`

**Comment by TomasTurina** [comment]
Will re-do.

**Pull request closed without merging**